mempool/src/core_mempool/transaction_store.rs

```diff
@@ -428,27 +430,38 @@ impl TransactionStore {
         self.track_indices();
     }
 
-    /// Read `count` transactions from timeline since `timeline_id`.
+    /// Read at most `count` transactions from timeline since `timeline_id`.
+    /// This method takes into account the max number of bytes per transaction batch.
     /// Returns block of transactions and new last_timeline_id.
     pub(crate) fn read_timeline(
         &self,
         timeline_id: u64,
         count: usize,
     ) -> (Vec<SignedTransaction>, u64) {
         let mut batch = vec![];
+        let mut batch_total_bytes = 0;
         let mut last_timeline_id = timeline_id;
+
+        // Add as many transactions to the batch as possible
         for (address, sequence_number) in self.timeline_index.read_timeline(timeline_id, count) {
             if let Some(txn) = self
                 .transactions
                 .get(&address)
                 .and_then(|txns| txns.get(&sequence_number))
             {
-                batch.push(txn.txn.clone());
-                if let TimelineState::Ready(timeline_id) = txn.timeline_state {
-                    last_timeline_id = timeline_id;
+                let transaction_bytes = txn.txn.raw_txn_bytes_len() as u64;
+                if (batch_total_bytes + transaction_bytes) > self.max_batch_bytes {
```

Comment:
saturating add?  Though I guess max batch bytes should never get close to that

Reply:
Aah, thanks!